File: Java/oopWeek1WednesdayAM/BankAccount.java
public class BankAccount {
    private String firstName;
    private String lastName;
    private Double accountBalance;

    public BankAccount(){
        this.firstName = "Untitled firstname";
        this.lastName = "Untitled lastname";
        this.accountBalance = 0.00;
    }

    //this is my constructor (similar to the def __init__() method in python)
    public BankAccount(String firstNameInput, String lastNameInput){
        this.firstName = firstNameInput;
        this.lastName = lastNameInput;
        this.accountBalance = 0.00;

    }

    public BankAccount(String firstNameInput, String lastNameInput, Double initialAmount){
        this.firstName = firstNameInput;
        this.lastName = lastNameInput;
        this.accountBalance = initialAmount;
    }

    public Double deposit(){
        this.accountBalance += 10;
        return this.accountBalance;
    }

    public Double deposit(Double amount){
        this.accountBalance += amount;
        return this.accountBalance;
    }


    //a getter to get the accountbalance
    public Double getAccountBalance(){
        return this.accountBalance;
    }

    //a setter allows us to modify private variables
    public void setAccountBalance(Double newBalance){
        this.accountBalance = newBalance;

    }

    //who to send moneyto (which account)
    // the amount you want to send
    public void transferMoney(BankAccount otherAccount, Double amountToSend){
        //the object that called the transferMoney method is referred to with the keyword "this"
        this.accountBalance -= amountToSend;
        otherAccount.accountBalance += amountToSend;
    }

    public String displayAccountInfo(){
        return String.format("Account Balance: %f. First Name: %s, Last Name: %s", this.accountBalance, this.firstName, this.lastName);
        // return "Account Balance: " + this.accountBalance + " First Name: " + this.firstName;
    }
}

File: Python/fundamentals/oop/BankAccount/bankaccount.py
class BankAccount:
	def __init__(self, int_rate, balance):
		self.account_balance = balance
		self.int_rate = int_rate

	def deposit(self, amount):
		self.account_balance += amount
		return self

	def withdraw(self, amount):
		if self.account_balance > amount:
			self.account_balance -= amount
			return self
		else:
			print("Insufficient funds: Charging a $5 fee")
			self.account_balance -= 5
			return self
			
	def display_account_info(self):
		print(f"Balance: ${self.account_balance}")
	
	def yield_interest(self):
		if self.account_balance > 0:
			self.account_balance *= self.int_rate
			return self


user1 = BankAccount(1.01, 0)
user2 = BankAccount(1.02, 1000)

user1.deposit(100).deposit(200).deposit(700).withdraw(1050).yield_interest().display_account_info()
user2.deposit(1000).deposit(100).withdraw(200).withdraw(300).withdraw(200).withdraw(200).yield_interest().display_account_info()

user1
user2